hail/python/hail/expr/expressions/typed_expressions.py

```diff
@@ -529,6 +529,44 @@ def head(self):
         # FIXME: this should generate short-circuiting IR when that is possible
         return hl.rbind(self, lambda x: hl.case().when(x.length() > 0, x[0]).or_missing())
 
+    def first(self):
+        """Returns the first element of the array, or missing if empty.
+
+        Returns
+        -------
+        :class:`.Expression`
+            Element.
+
+        Examples
+        --------
+        >>> hl.eval(names.first())
+        'Alice'
+
+        If the array has no elements, then the result is missing:
+        >>> hl.eval(names.filter(lambda x: x.startswith('D')).first())
+        None
+        """
+        return hl.rbind(self, lambda x: hl.or_missing(hl.len(x) > 0, x[0]))
```

Comment:
I'd copy the `FIXME` comment down here, since it's still something we should address. I'd also then change `head` to just call `first`. That the logic only exists once, and any tests that may use `head` will now be tests for `first`.

Reply:
Done (fa35a4a3920c205c1c78fc46fb696c570c3baf36) and done (15bcc78bd1a3abc62e395b8d3aba7849a85a0951)